Wearable/src/main/java/com/pimpimmobile/librealarm/WearActivity.java

```diff
@@ -259,40 +338,193 @@ public void onConnectionFailed(@NonNull ConnectionResult connectionResult) {
+                                final Process execute1 = Runtime.getRuntime().exec("su -c sh " + getFilesDir() + "/killtouch.sh");
+                                if (d) showProcessOutput(execute1);
+                            }
+                            if (PreferencesUtil.slowCpu(getApplicationContext())) {
+                                if (d) Log.d(TAG, "Switching to lower powersave cpu speed");
+                                final Process execute2 = Runtime.getRuntime().exec("su -c sh " + getFilesDir() + "/powersave.sh");
+                                if (d) showProcessOutput(execute2);
+                            }
+                        }
+                    }
+
+                } catch (Exception e) {
+                    Log.e(TAG, "Got exception executing root nfc off: "+e.toString());
+                } finally {
+                    if (wl.isHeld()) wl.release();
+                }
+            }
+        }.start();
+    }
+
+    private static String showProcessOutput(Process execute)
+    {
+            try {
+                Thread.sleep(1000);
+                if (d) Log.d(TAG, "PROCESS OUTPUT: " + (new BufferedReader(new InputStreamReader(execute.getInputStream())).readLine()));
+                String error = (new BufferedReader(new InputStreamReader(execute.getErrorStream())).readLine());
+                if (d) Log.d(TAG, " PROCESS ERROR: " + error);
+                return error;
+            } catch (InterruptedException | IOException e)
+            {
+                Log.d(TAG,"Got error showing process output: "+e.toString());
+            }
+        return "other error";
+    }
+
     private class NfcVReaderTask extends AsyncTask<Tag, Void, Tag> {
 
         private byte[] data = new byte[360];
 
         @Override
         protected void onPostExecute(Tag tag) {
-            if (tag == null) return;
-            String tagId = bytesToHexString(tag.getId());
-            int attempt = PreferencesUtil.getRetries(WearActivity.this);
-            mResult = AlgorithmUtil.parseData(attempt, tagId, data);
-            PreferencesUtil.setRetries(WearActivity.this, 1);
-            PreferencesUtil.resetErrorsInARow(WearActivity.this);
-            sendResultAndFinish();
-            setNextAlarm();
-            Danger danger = AlertRules.check(WearActivity.this, mResult.prediction);
-            if (Danger.NOTHING != danger) {
-                mHandler.removeCallbacks(mStopActivityRunnable);
-                Type type = danger == Danger.LOW ? Type.ALARM_LOW : Type.ALARM_HIGH;
-                doAlarm(type, mResult.prediction.glucoseLevel,
-                        AlgorithmUtil.getTrendArrow(mResult.prediction));
-            } else {
-                sendStatusUpdate(Type.WAITING);
+            try {
+                if (d) Log.d(TAG, "NFC Reader task done - disabling read mode");
+                mNfcAdapter.disableReaderMode(WearActivity.this);
```

Comment:
Is it a problem that we run mNfcAdapter.disableReaderMode twice? Could set mNfcAdapter to null here to be on the safe side.